File: 品味Java的21種設計模式/dp_java/bridge/example6/SpecialUrgencyMessage.java
package cn.javass.dp.bridge.example6;
/**
 * 特急消息
 */
public class SpecialUrgencyMessage extends AbstractMessage{

	public SpecialUrgencyMessage(MessageImplementor impl) {
		super(impl);
	}

	public void hurry(String messageId) {
		//执行催促的业务，发出催促的信息
	}

	public void sendMessage(String message, String toUser) {
		message = "特急："+message;
		super.sendMessage(message, toUser);
		//还需要增加一条待催促的信息
	}
}

File: 品味Java的21種設計模式/dp_cpp/bridge/example6/SpecialUrgencyMessage.cpp
#include "SpecialUrgencyMessage.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace bridge
			{
				namespace example6
				{

					SpecialUrgencyMessage::SpecialUrgencyMessage(MessageImplementor *impl) : AbstractMessage(impl)
					{
					}

					void SpecialUrgencyMessage::hurry(std::string messageId)
					{
						//执行催促的业务，发出催促的信息
					}

					void SpecialUrgencyMessage::sendMessage(std::string message, std::string toUser)
					{
						message = "特急："+message;
						AbstractMessage::sendMessage(message, toUser);
						//还需要增加一条待催促的信息
					}
				}
			}
		}
	}
}
